expression/builtin_math.go

```diff
@@ -1030,7 +1030,7 @@ func (c *randFunctionClass) getFunction(ctx sessionctx.Context, args []Expressio
 			// The behavior same as MySQL.
 			seed = 0
 		}
-		sig = &builtinRandSig{bt, &sync.Mutex{}, NewWithSeed(seed)}
+		sig = &builtinRandSig{bt, utilMath.NewWithSeed(seed)}
```

Comment:
LGTM, but I'm not sure about removing the mutex that was previously used. Can you explain in the notes why it is safe?

Reply:
Added it in https://github.com/pingcap/tidb/pull/29936/files#diff-f499163276ef45b95eb6434def8e512154988d9dad6c82a6342e8244514c3da8R39. The `mysqlRng` includes the lock.